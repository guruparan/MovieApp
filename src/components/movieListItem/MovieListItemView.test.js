
import renderer from 'react-test-renderer';
import MovieListItem from './MovieListItemView';
import React from 'react';

test('renders correctly', () => {
    const tree = renderer.create(<MovieListItem movie={{
        release_date: '2020-01-01'
    }} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Valid movie title', () => {
    const testRenderer = renderer.create(<MovieListItem movie={{
        title:'Test',
        release_date: '2020-01-01'
    }} />);
    const testInstance = testRenderer.root;
    
    expect(testInstance.findByProps({testID: "movieTitle"}).props.children.join()).toBe('Test, (2020)');
});