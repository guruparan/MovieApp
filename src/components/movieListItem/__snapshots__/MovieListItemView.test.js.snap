// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#D5D3D3",
      "borderBottomLeftRadius": 40,
      "borderRadius": 5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "marginVertical": 8,
      "opacity": 1,
      "padding": 10,
      "width": "95%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../images/poster.png",
      }
    }
    style={
      Object {
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "borderRadius": 5,
        "height": 80,
        "marginRight": 10,
        "width": 58,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 5,
          "borderRadius": 15,
          "borderTopWidth": 0,
          "bottom": -5,
          "height": 30,
          "justifyContent": "center",
          "left": 25,
          "position": "absolute",
          "width": 30,
        },
        Object {
          "borderColor": "grey",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
      testID="movieTitle"
    >
       (2020)
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginTop": 10,
        }
      }
    >
      Released on 
      2020-01-01
    </Text>
  </View>
</View>
`;
